distributed/cli/tests/test_dask_worker.py

```diff
@@ -152,7 +119,8 @@ async def test_no_reconnect(c, s, nanny):
             "--no-reconnect",
             nanny,
             "--no-dashboard",
-        ]
+        ],
+        flush_output=False,
```

Comment:
Why not `flush_output=True` here and remove the `await to_thread(worker.communicate, timeout=5)`? Maybe there's a subtle difference in timing I'm missing.

Reply:
This and another test were abusing ``Proc.communicate`` to just wait for the subprocess to finish. I changed them now.